broker/machinebroker/server/event_list.go

```diff
@@ -12,35 +12,56 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	computev1alpha1 "github.com/ironcore-dev/ironcore/api/compute/v1alpha1"
 	"github.com/ironcore-dev/ironcore/broker/machinebroker/apiutils"
 	iri "github.com/ironcore-dev/ironcore/iri/apis/machine/v1alpha1"
 )
 
-const InvolvedObjectName = "involvedObject.name"
+const (
+	InvolvedObjectKind               = "Machine"
+	InvolvedObjectKindSelector       = "involvedObject.kind"
+	InvolvedObjectAPIversionSelector = "involvedObject.apiVersion"
+)
 
-func (s *Server) listEvents(ctx context.Context, machineID string) ([]*iri.Event, error) {
+func (s *Server) listEvents(ctx context.Context) ([]*iri.Event, error) {
 	machineEventList := &v1.EventList{}
 	selectorField := fields.Set{}
-	selectorField[InvolvedObjectName] = machineID
+	selectorField[InvolvedObjectKindSelector] = InvolvedObjectKind
+	selectorField[InvolvedObjectAPIversionSelector] = computev1alpha1.SchemeGroupVersion.String()
 
 	if err := s.cluster.Client().List(ctx, machineEventList,
 		client.InNamespace(s.cluster.Namespace()), client.MatchingFieldsSelector{Selector: selectorField.AsSelector()},
 	); err != nil {
-		return nil, fmt.Errorf("error listing machine events: %w", err)
+		return nil, err
 	}
 
 	var iriEvents []*iri.Event
+	var eventTime int64
 	for _, machineEvent := range machineEventList.Items {
-		iriEvent := &iri.Event{
-			Metadata: apiutils.GetIRIObjectMetadata(&machineEvent.ObjectMeta),
-			Spec: &iri.EventSpec{
-				Reason:    machineEvent.Reason,
-				Message:   machineEvent.Message,
-				Type:      machineEvent.Type,
-				EventTime: machineEvent.FirstTimestamp.Unix(),
-			},
+		ironcoreMachine, err := s.getIronCoreMachine(ctx, machineEvent.InvolvedObject.Name)
+		if err != nil {
+			continue
+		} else {
+			machineObjectMetadata, err := apiutils.GetObjectMetadata(&ironcoreMachine.ObjectMeta)
+			if err != nil {
+				continue
+			} else {
+				eventTime = machineEvent.LastTimestamp.Unix()
+				if eventTime < 0 {
```

Comment:
Is this check correct? Can you verify if the time is negative if it's not set?

Reply:
I just rechecked k8s event recorder LastTimestamp is always set while generating event , so directly using it
https://github.com/kubernetes/kubernetes/blob/0fa4b9e14cba6342cfaaea4db793e36e4b8ae20d/staging/src/k8s.io/client-go/tools/record/event.go#L500